### PR TITLE
Make JSON files more human readable

### PR DESCRIPTION
**Pull request by janeyx99**
Prettifies JSON files .pytorch-test-times and .pytorch-slow-tests so that not everything is on one single line.

This is of slightly more importance as generated  .pytorch-slow-tests ends up getting stored in our test-infra repo ([example](https://github.com/pytorch/test-infra/commit/ad9cd8756589aa28b6f2817caaf074ca93e76cd9)), and it is nice to not have that lil red symbol at the end.

### CODE REVIEW

File: test/run_test.py [hunk 1]
```diff
@@ -1017,7 +1017,9 @@ def export_S3_test_times(test_times_filename: str, test_times: Dict[str, float])
         print(f'Overwriting existent file: {test_times_filename}')
     with open(test_times_filename, 'w+') as file:
         job_times_json = get_job_times_json(test_times)
-        json.dump(job_times_json, file)
+        json.dump(job_times_json, file, indent='    ', separators=(',', ': '))
+    with open(test_times_filename, 'a') as file:
```
Comment (samestep) [inline test/run_test.py:1021]: Why do you close the file and then reopen it again to append a newline?
```suggestion
```

File: tools/export_slow_tests.py [hunk 2]
```diff
@@ -43,6 +43,8 @@ def export_slow_tests(filename: str) -> None:
     with open(filename, 'w+') as file:
         slow_test_times: Dict[str, float] = filter_slow_tests(get_test_case_times())
         json.dump(slow_test_times, file, indent='    ', separators=(',', ': '))
+    with open(filename, 'a') as file:
```
Comment (samestep) [inline tools/export_slow_tests.py:46]: Same here
```suggestion
```

**Review by samestep — APPROVED** [review]

File: test/run_test.py (hunk 1 above)
Comment (janeyx99) [reply test/run_test.py:1021]: I was under the impression that w+ and file.write() would overwrite the whole thing again, but now I realize I misread the docs :c

**Review by walterddr — APPROVED** [review]
you can also use [pprint](https://docs.python.org/3/library/pprint.html), it will avoid the need to append the last newline I think (test on terminal, not file though)

**Pull request closed without merging**